clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -134,6 +136,173 @@ bool SYCL::shouldDoPerObjectFileLinking(const Compilation &C) {
                               /*default=*/true);
 }
 
+// Return whether to use native bfloat16 library.
+static bool selectBfloatLibs(const llvm::Triple &Triple, const Compilation &C,
+                             bool &UseNative) {
+  const llvm::opt::ArgList &Args = C.getArgs();
+  bool NeedLibs = false;
+
+  // spir64 target is actually JIT compilation, so we defer selection of
+  // bfloat16 libraries to runtime. For AOT we need libraries, but skip
+  // for Nvidia.
+  NeedLibs =
+      Triple.getSubArch() != llvm::Triple::NoSubArch && !Triple.isNVPTX();
+  UseNative = false;
+  if (NeedLibs && Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen &&
+      C.hasOffloadToolChain<Action::OFK_SYCL>()) {
+    ArgStringList TargArgs;
+    auto ToolChains = C.getOffloadToolChains<Action::OFK_SYCL>();
+    // Match up the toolchain with the incoming Triple so we are grabbing the
+    // expected arguments to scrutinize.
+    for (auto TI = ToolChains.first, TE = ToolChains.second; TI != TE; ++TI) {
+      llvm::Triple SYCLTriple = TI->second->getTriple();
+      if (SYCLTriple == Triple) {
+        const toolchains::SYCLToolChain *SYCLTC =
+            static_cast<const toolchains::SYCLToolChain *>(TI->second);
+        SYCLTC->TranslateBackendTargetArgs(Triple, Args, TargArgs);
+        break;
+      }
+    }
+
+    auto checkBF = [=](std::string &Dev) {
+      static const std::regex BFFs("pvc.*|ats.*");
```

Comment:
nit: We can use[`llvm::StringRef::starts_with()`](https://llvm.org/doxygen/classllvm_1_1StringRef.html#a2cd8968ff703aaeb395dcd63f6805ff1) function for this until C++20 turns on.

Reply:
I have updated to use `starts_with`